src/pages/CommandView/dynamic-form/dynamic-form-helpers/get-on-change-functions.ts

```diff
@@ -0,0 +1,620 @@
+        const target = event.target
+
+        if ('value' in target && 'name' in target) {
+          const { value, name } = target as EventTarget & {
+            value: string
+            name: string
+          }
+
+          // update the form library's context object
+          const newContextParameters = {
+            ...(context.values['parameters'] as ObjectWithStringKeys),
+            [name]: value,
+          }
+          context.setFieldValue('parameters', newContextParameters)
+
+          // update our tracked 'model' state
+          stateManager.model.set((prev) => {
+            return {
+              ...prev,
+              parameters: {
+                ...prev.parameters,
+                [name]: value,
+              },
+            }
+          })
+
+          if (canUpdate()) updateDynamic()
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Get an 'onChange' function for a parameter which is a must choose dependency
+ * of a dynamic choice parameter
+ *
+ * @param tag - the name of the parameter or parameter-like that we're building
+ *     a function for
+ * @param shouldUpdate - an array of the dynamic choice parameters that should
+ *     be updated by a change in this parameter
+ * @param dynamicChoices - an object keyed by the name of the dynamic choice
+ *     parameters, whose values are the properties necessary to build a request
+ *     to update the dynamic choice values
+ * @param mustChoose - a set of the names of parameters that must be set to
+ *     an initial value before an update request can be executed
+ * @param stateManager - the getters and setters of the state we need to keep
+ *     track of
+ * @returns - a function for use in individual components to get an 'onChange'
+ *     function; this function takes a FormikContextType, the execute function
+ *     from an invocation of 'useAxios', an 'onError' function in case
+ *     anything goes wrong when running 'execute' and an 'authEnabled' boolean
+ *     indicating whether authentication is enabled and returns a function to
+ *     be used for 'onChange' in the component
+ */
+const getOnChangeForMustChoose = (
+  tag: string,
+  shouldUpdate: string[],
+  dynamicChoices: Record<string, DynamicProperties>,
+  mustChoose: Set<string>,
+  stateManager: DynamicChoicesStateManager,
+): OnChangeGetterFunction => {
+  /* at this time, we assume every 'must choose' is not actually a proper
+     parameter of the request -- e.g. 'instance_name' */
+  const otherMustChoose = [...mustChoose].filter((x) => x !== tag)
+
+  return <T extends Record<string, unknown>>(
+    context: FormikContextType<T>,
+    execute: (
+      config?: AxiosRequestConfig<RequestTemplate>,
+    ) => AxiosPromise<Request>,
+    onError: (e: AxiosError<T>) => void,
+    authEnabled: boolean,
+  ) => {
+    const canUpdate = () => {
+      return (
+        otherMustChoose.length === 0 ||
+        otherMustChoose
+          .map((x) => stateManager.ready.get()[x].ready)
```

Comment:
can you just do .every() here instead of mapping?

Reply:
Fixed.